### PR TITLE
Add VSB Finalizers and catch VSB errors

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

File: controllers/cleanup.go [hunk 1]
```diff
@@ -29,16 +29,18 @@ var cleanupVSRTypes = []client.Object{
...
 		return false, err
 	}
 
-	// make sure VSB is completed before deleting resources
-	if vsb.Status.Phase != volsnapmoverv1alpha1.SnapMoverVolSyncPhaseCompleted {
+	// make sure VSB is completed or failed before deleting resources AND VSB has not been deleted
+	if (vsb.Status.Phase != volsnapmoverv1alpha1.SnapMoverVolSyncPhaseCompleted ||
+		vsb.Status.Phase != volsnapmoverv1alpha1.SnapMoverBackupPhaseFailed ||
+		vsb.Status.Phase != volsnapmoverv1alpha1.SnapMoverBackupPhasePartiallyFailed) &&
+		vsb.DeletionTimestamp == nil {
```
Comment (dymurray) [inline controllers/cleanup.go:43]: I should change this to `vsb.DeletionTimestamp.IsZero()`

File: controllers/cleanup.go [hunk 2]
```diff
@@ -65,10 +67,12 @@ func (r *VolumeSnapshotBackupReconciler) CleanBackupResources(log logr.Logger) (
 	// }
 
 	// Update VSB status as completed
-	vsb.Status.Phase = volsnapmoverv1alpha1.SnapMoverBackupPhaseCompleted
-	err := r.Status().Update(context.Background(), &vsb)
-	if err != nil {
-		return false, err
+	if vsb.DeletionTimestamp == nil {
```
Comment (dymurray) [inline controllers/cleanup.go:70]: Change to use `IsZero()`

File: controllers/volumesnapshotbackup_controller.go [hunk 3]
```diff
@@ -95,14 +100,40 @@ func (r *VolumeSnapshotBackupReconciler) Reconcile(ctx context.Context, req ctrl
...
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		controllerutil.RemoveFinalizer(&vsb, dmFinalizer)
+		err = r.Update(ctx, &vsb)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Clean up successful")
```
Comment (eemcmullan) [inline controllers/volumesnapshotbackup_controller.go:133]: should we keep this log in? Debug log?

File: api/v1alpha1/volumesnapshotbackup_types.go [hunk 4]
```diff
@@ -72,6 +72,10 @@ const (
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 //+kubebuilder:resource:path=volumesnapshotbackups,shortName=vsb
+// +kubebuilder:printcolumn:name="PVC Name",type=string,JSONPath=".status.sourcePVCData.name"
```
[the remark this answers is not in the document]
Comment (shubham-pampattiwar) [reply api/v1alpha1/volumesnapshotbackup_types.go:75]: I believe the printer columns will show empty data

File: config/crd/bases/datamover.oadp.openshift.io_volumesnapshotbackups.yaml [hunk 5]
```diff
@@ -17,7 +17,20 @@ spec:
...
+      name: PVC Name
+      type: string
+    - jsonPath: .spec.volumeSnapshotContent.name
+      name: VolumeSnapshotContent
+      type: string
+    - jsonPath: .status.phase
+      name: Phase
+      type: string
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
```
Comment (kaovilai) [inline config/crd/bases/datamover.oadp.openshift.io_volumesnapshotbackups.yaml:31]: Why is use `Age` which is different from `creationTimestamp`?

If it is truly a type of date.. it won't be displaying age (number of days/months/years) but rather a date (date/month/year) that VSB was created on.

**Review by kaovilai — CHANGES_REQUESTED** [review]

**Review by eemcmullan — APPROVED** [review]

**Comment by dymurray** [comment]
LGTM

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by kaovilai — APPROVED** [review]

**Pull request merged**